### PR TITLE
Experiment with Turbo

### PR DESCRIPTION
**Pull request by djfarrelly**
## Description

Use Turbo pack and repo to run the Next.js dev server and build in production.

## Motivation
Pushing a spike from a couple weeks ago to improve the build times during dev and deployment. Also a secondary goal of reducing the memory footprint during local dev.

## Type of change (choose one)
- [x] Chore (refactors, upgrades, etc.)
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] Security fix (non-breaking change that fixes a potential vulnerability)
- [ ] Docs
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)

## Checklist
- [ ] I've linked any associated issues to this PR.
- [ ] I've tested my own changes.

*[Check our Pull Request Guidelines](https://github.com/inngest/inngest/blob/main/docs/PULL_REQUEST_GUIDELINES.md)*

### CODE REVIEW

**Comment by djfarrelly** [comment]
Currently running turbo pack (`next dev --turbo`) fails as it can't seem to find the packages workspaces and components: 
image

File: ui/apps/dashboard/package.json [hunk 1]
```diff
@@ -111,7 +111,7 @@
     "vercel": "32.4.1"
   },
   "engines": {
-    "node": "18.x",
+    "node": "20.x",
```
Comment (djfarrelly) [inline ui/apps/dashboard/package.json:114]: Turbo requires Node 20.x which is already LTS, so I think it's good to use it.

File: ui/turbo.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    },
+    "type-check": {
+      "dependsOn": ["^type-check"]
+    },
+    "dev": {
+      "persistent": true,
+      "cache": false
+    }
+  }
+}
```
Comment (djfarrelly) [inline ui/turbo.json:16]: This is added for the turbo repo work which can build and cache files more quickly across a monorepo. Mostly, I was interested in Turbo pack for local dev which seems to work alright without this file.

File: ui/packages/tsconfig/base.json [hunk 3]
```diff
@@ -13,7 +13,10 @@
     "declarationMap": true,
     "esModuleInterop": true,
     "verbatimModuleSyntax": true,
-    "skipLibCheck": true
+    "skipLibCheck": true,
+    "paths": {
+      "@inngest/components/*": ["../components/src/*"]
+    }
```
Comment (djfarrelly) [inline ui/packages/tsconfig/base.json:19]: This is needed for Turbo pack as per their docs. Before this change, the main `dashboard` app would be able to load the component (e.x. `Alert.tsx`) from the package (I believe due to the `package.json`'s `dependencies`), but it wouldn't be able to load the nested deps from within that component, e.g. `@inngest/components/utils/classNames` until this was added.

**Review by jacobheric — APPROVED** [review]

**Pull request merged**